### ISSUE TITLE
Lack of Root exports causing TypeScript portability issues

### ISSUE THREAD

**Issue by manspaniel** [chakra-ui/ark#1740]
### Description

When re-exporting a root component (in my case, `Select.Root`), I'm receiving the following TypeScript issue.

```
The inferred type of 'SelectRoot' cannot be named without a reference to '../../../../node_modules/@ark-ui/react/select/select'. This is likely not portable. A type annotation is necessary.
```

_Context: I'm using Park UI, and defining my base components in a sub-package of Monorepo, for consumption in other packages + a Next.js app_

I'm noticing in the `@ark-ui/react/select/index.dts`, that all sub-components are being exported, except for `SelectComponent` (the root component).

There is no way to access the type of this component as a consumer of Ark.

Adding `export type { SelectComponent }` to `@ark-ui/react/select/index.dts` solves the issue entirely.

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/s/ark-ui-root-export-rqsplw?file=/src/Select.tsx

### Steps to Reproduce

1. Using `tsconfig.ts`
```
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "Bundler",
```
3. Re-export Select.Root

### Ark UI Version

1.0.1

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

### Additional Information

Swapping to `moduleResolution: "node"` silences the error also, but if the package is then consumed by a Next.js application, which defaults to `moduleResolution: "bundler"`, then Intellisense does not work for the Root component, with no warnings.

I believe this could be related to [park-ui#87](https://github.com/cschroeter/park-ui/issues/87)

Screenshot:

<img width="691" alt="image" src="https://github.com/chakra-ui/ark/assets/1719776/6d437f19-47bf-4c4b-839c-9b9b856d8e1d">

**Comment by cschroeter** [comment]
@manspaniel 
Thanks for taking your time to create this issue. Could you check if the Combobox is also affected in your setup? Both components are using a generic interface.

**Comment by manspaniel** [comment]
@cschroeter 
Thanks for taking the time to create this library! I'm really loving the direction Panda + Park + Ark. It's refreshing to see some strong competition with Radix.

Combobox seems to suffer from the same issue: [same codesandbox](https://codesandbox.io/s/ark-ui-root-export-rqsplw?file=/src/ComboBox.tsx)

**Comment by manspaniel** [comment]
Again, changing to `moduleResolution: "node"` silences the warning, but causes downstream Intellisense issues when a consuming package uses `moduleResolution: "bundler"`.

I can work around the issue by adding the following to my UI package:

```diff
export type SelectProps<T extends Ark.CollectionItem> = Ark.SelectProps<T> & SelectVariantProps;

+ type SelectRoot = <T extends Ark.CollectionItem>(
+   props: SelectProps<T>
+ ) => JSX.Element;

- const SelectRoot = withProvider(styled(Ark.Select.Root), "root");
+ const SelectRoot = withProvider(styled(Ark.Select.Root as SelectRoot), "root");
```
(Modified from [Park UI's Select](https://park-ui.com/docs/panda/components/select))

**Comment by gurkerl83** [comment]
We changed also the nodeResolution to node in our project. I have several next apps and I only use node for nodeResolution without problems. It seems the value bundler in your NextJS app was added by creating a next app with `create-next-app`. No idea why `create-next-app` defaults bundler in the first place.

**Comment by cschroeter** [comment]
This will be resolved in v2.0
